api/types/swarm/service.go

```diff
@@ -17,6 +17,10 @@ type Service struct {
 	// listing all tasks for a service, an operation that could be
 	// computation and network expensive.
 	ServiceStatus *ServiceStatus `json:",omitempty"`
+
+	// JobStatus is the status of a Service which is in one of ReplicatedJob or
+	// GlobalJob modes. It is absent on Replicated and Global services.
+	JobStatus *JobStatus `json:",omitempty"`
```

Comment:
How come this is not part of `ServiceStatus`?

Reply:
`ServiceStatus` is only available when doing a List operation of Services, where it is used to store the result of server-side computation of task states for the service. That is, `ServiceStatus` is an optimization of a common operation (counting desired/running task counts). `JobStatus` is an intrinsic part of the `Service` object, not derived or computed.